test/pstl_offload/memory/allocation_utils.h

```diff
@@ -0,0 +1,119 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ALLOCATION_UTILS_H
+#define _ALLOCATION_UTILS_H
+
+#define _ONEDPL_PSTL_OFFLOAD_TOP_LEVEL
+#include <new>
+#include <cstdlib>
+#if __linux__
+#include <malloc.h>
+#endif
+#include "support/utils.h"
+#undef _ONEDPL_PSTL_OFFLOAD_TOP_LEVEL
```

Comment:
what if we just extract these lines to a separate header (or just include those in the file where we have ``perform_system_allocations`` functions? In this case utils would just be utils and the decision about memory redirection would be done by translation unit.

I am not sure if it would be better or not but I am suggesting to experiment with that.

Reply:
Done.